commit 643f82e32f14faf0d0944c804203a6681b6b0a1e
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Mon Jul 12 14:46:43 2010 +0200

    cfg80211: ignore spurious deauth
    
    Ever since mac80211/drivers are no longer
    fully in charge of keeping track of the
    auth status, trying to make them do so will
    fail. Instead of warning and reporting the
    deauthentication to userspace, cfg80211 must
    simply ignore it so that spurious
    deauthentications, e.g. before starting
    authentication, aren't seen by userspace as
    actual deauthentications.
    
    Cc: stable@kernel.org
    Reported-by: Paul Stewart <pstew@google.com>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/wireless/mlme.c b/net/wireless/mlme.c
index 9f95354f859f..e74a1a2119d3 100644
--- a/net/wireless/mlme.c
+++ b/net/wireless/mlme.c
@@ -44,10 +44,10 @@ void cfg80211_send_rx_auth(struct net_device *dev, const u8 *buf, size_t len)
 		}
 	}
 
-	WARN_ON(!done);
-
-	nl80211_send_rx_auth(rdev, dev, buf, len, GFP_KERNEL);
-	cfg80211_sme_rx_auth(dev, buf, len);
+	if (done) {
+		nl80211_send_rx_auth(rdev, dev, buf, len, GFP_KERNEL);
+		cfg80211_sme_rx_auth(dev, buf, len);
+	}
 
 	wdev_unlock(wdev);
 }